2007-11-07  Jim Meyering  <meyering@redhat.com>
            Bruno Haible  <bruno@clisp.org>

	* write-catalog.c (msgdomain_list_print): Fix open() call.

--- a/gettext-tools/src/write-catalog.c	7 Oct 2007 19:35:31 -0000	1.4
+++ b/gettext-tools/src/write-catalog.c	7 Nov 2007 11:43:15 -0000
@@ -1,5 +1,5 @@
 /* GNU gettext - internationalization aids
-   Copyright (C) 1995-1998, 2000-2006 Free Software Foundation, Inc.
+   Copyright (C) 1995-1998, 2000-2007 Free Software Foundation, Inc.
 
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -220,7 +220,9 @@
       /* Open the output file.  */
       if (!to_stdout)
 	{
-	  fd = open (filename, O_WRONLY | O_CREAT);
+	  fd = open (filename, O_WRONLY | O_CREAT | O_TRUNC,
+		     /* 0666 in portable POSIX notation: */
+		     S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH);
 	  if (fd < 0)
 	    {
 	      const char *errno_description = strerror (errno);
